### PR TITLE
Allow shoppers remove stored cards for session integration

### PR DESCRIPTION
**Pull request by araratthehero**
## Description
- Now merchants do not have to override `onRemoveStoredPaymentMethod` to support removing stored payment methods for session integration.
- A new toggle has been added to the settings page of the example app to allow removing stored payment methods.
- `SessionConfiguration` object has been created to optimize `SessionParams` and keep configuration related params separate.

To be done:
- Align error messages with other platforms.

## Checklist 
- [x] Code is unit tested
- [x] Changes are tested manually
- [x] Add relevant labels to PR  

COAND-835

### CODE REVIEW

File: drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/model/DropInParamsMapper.kt [hunk 1]
```diff
@@ -39,4 +43,10 @@ internal class DropInParamsMapper {
     fun getShopperLocale(checkoutConfiguration: CheckoutConfiguration, sessionParams: SessionParams?): Locale? {
         return checkoutConfiguration.shopperLocale ?: sessionParams?.shopperLocale
     }
+
+    private fun getIsRemovingStoredPaymentMethodsEnabled(
+        dropInConfiguration: DropInConfiguration?,
+        sessionParams: SessionParams?
+    ) = dropInConfiguration?.isRemovingStoredPaymentMethodsEnabled
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/model/DropInParamsMapper.kt:50]: This code allows the client side configuration to override the sessions configuration, is that intended?

File: sessions-core/src/test/java/com/adyen/checkout/sessions/core/internal/SessionInteractorTest.kt [hunk 2]
```diff
@@ -707,6 +708,39 @@ internal class SessionInteractorTest(
         }
     }
 
+    @Nested
+    @DisplayName("when an update payment methods call is requested and")
```
Comment (jreij) [inline sessions-core/src/test/java/com/adyen/checkout/sessions/core/internal/SessionInteractorTest.kt:712]: I think the test names need to be fixed

File: sessions-core/src/main/java/com/adyen/checkout/sessions/core/internal/ui/model/SessionParamsFactory.kt [hunk 3]
```diff
@@ -34,16 +35,19 @@ object SessionParamsFactory {
...
-            installmentConfiguration = SessionInstallmentConfiguration(
-                installmentOptions = sessionSetupConfiguration?.installmentOptions?.map {
-                    it.key to SessionInstallmentOptionsParams(
-                        plans = it.value?.plans,
-                        preselectedValue = it.value?.preselectedValue,
-                        values = it.value?.values,
-                    )
-                }?.toMap(),
-                showInstallmentAmount = sessionSetupConfiguration?.showInstallmentAmount,
+            configuration = SessionConfiguration(
```
Comment (jreij) [inline sessions-core/src/main/java/com/adyen/checkout/sessions/core/internal/ui/model/SessionParamsFactory.kt:38]: The field is marked as nullable but it doesn't look like it can be null

Comment (araratthehero) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/core/internal/ui/model/SessionParamsFactory.kt:38]: The way we have it implemented it is not null, but I kept it null in the `SessionConfiguration` to keep it consistent with other fields in `SessionParams` and have them all nullable. Which also allows us to pass null in tests, where we do not need to create an `SessionConfiguration` object and pass null for all of it's arguments.

Though I do not mind making it non-nullable. Let me know what you think.

File: sessions-core/src/test/java/com/adyen/checkout/sessions/core/internal/SessionInteractorTest.kt (hunk 2 above)
Comment (araratthehero) [reply sessions-core/src/test/java/com/adyen/checkout/sessions/core/internal/SessionInteractorTest.kt:712]: Good catch!

Comment (jreij) [reply sessions-core/src/test/java/com/adyen/checkout/sessions/core/internal/SessionInteractorTest.kt:712]: Yes but these tests are not about "when an update payment methods call is requested and" but about removing a stored payment methods

Comment (araratthehero) [reply sessions-core/src/test/java/com/adyen/checkout/sessions/core/internal/SessionInteractorTest.kt:712]: Ahh I see. Totally missed that and confidently copied that part. 😅

**Comment by jreij** [comment]
@araratthehero I will review this again after you remove the draft status and rebase 🙏

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/model/SessionParams.kt [hunk 4]
```diff
@@ -12,10 +12,16 @@ import androidx.annotation.RestrictTo
 import com.adyen.checkout.components.core.Amount
 import java.util.Locale
 
+/**
+ * Object that holds values set during sessions setup call.
+ * [SessionParams] values should always have higher priority than values set in client side configurations. Otherwise
```
Comment (jreij) [inline components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/model/SessionParams.kt:17]: This is correct for most attributes but there are exceptions. For example `shopperLocale` can be overridden by a client side configuration

File: drop-in/src/main/java/com/adyen/checkout/dropin/SessionDropInService.kt [hunk 5]
```diff
@@ -222,6 +223,27 @@ open class SessionDropInService : BaseDropInService(), SessionDropInServiceInter
...
+            val result = sessionInteractor.removeStoredPaymentMethod(storedPaymentMethodId)
+
+            val serviceResult = when (result) {
+                SessionCallResult.RemoveStoredPaymentMethod.Successful ->
+                    RecurringDropInServiceResult.PaymentMethodRemoved(
+                        storedPaymentMethodId,
+                    )
+
+                is SessionCallResult.RemoveStoredPaymentMethod.Error -> RecurringDropInServiceResult.Error(
+                    errorDialog = ErrorDialog(),
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/SessionDropInService.kt:238]: What if we explicitly set the error message here even if it will match the default one? I'm thinking it might be tough for someone to read this line and figure out why aren't we setting any error message before finding out it will fall back to the default one anyway

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/model/SessionParams.kt (hunk 4 above)
Comment (araratthehero) [reply components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/model/SessionParams.kt]: Adjusted it a bit to give more context

File: RELEASE_NOTES.md [hunk 6]
```diff
@@ -82,3 +82,5 @@ CheckoutConfiguration(
 - When `CheckoutSessionProvider.createSession` to create a `CheckoutSession`, you can pass the `environment` and `clientKey` instead of the whole configuration.
 - In drop-in all actions will start in expanded mode
 - When using the Google Pay component, it is no longer necessary to manually import the `3ds2` module to handle transactions that require a native 3DS2 challenge. 
+- Removing stored payment methods are being handled internally if you are using Sessions integration. You do not need to override `onRemoveStoredPaymentMethod` function anymore.
+- Default api error message for drop-in changed from `Error sending payment. Please try again.` to `An unknown error occurred`.
```
Comment (jreij) [inline RELEASE_NOTES.md:86]: This can be any error that the merchant faced, we can't assume it's API. So maybe we can sat if you use `DropInServiceResult.Error` without specifying an error message, the default has changed?

Comment (araratthehero) [reply RELEASE_NOTES.md:86]: That is better indeed

**Review by jreij — APPROVED** [review]
